src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SessionStateTempDataProvider.cs

```diff
@@ -59,16 +59,10 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (!IsSessionEnabled(context))
-            {
-                // Session middleware is not enabled. No-op
-                return null;
-            }
-
             var session = context.Session;
```

Comment:
I seem to remember that detail from a previous PR. Do we have existing tests for this?